### PR TITLE
Memory leak select

### PR DESCRIPTION
**Pull request by Rincelent**
Correct a small memory leak happening when destroying select widget

### CODE REVIEW

File: src/composite/select.js [hunk 1]
```diff
@@ -325,8 +326,9 @@ var Select = Widget.$extend({
      * @method destroy
      */
     destroy: function () {
-        this.__popupMenu.destroy();
-        this.$super();
+        if (this.__displayValue) this.__displayValue.destroy()
+        if(this.__popupMenu) this.__popupMenu.destroy()
```
Comment (Rincelent) [inline src/composite/select.js:330]:
```suggestion
        if (this.__popupMenu) this.__popupMenu.destroy()
```

**Review by robincartier — APPROVED** [review]

**Review by aardvarksaurus — APPROVED** [review]

File: src/composite/select.js [hunk 2]
```diff
@@ -325,8 +326,9 @@ var Select = Widget.$extend({
      * @method destroy
      */
     destroy: function () {
-        this.__popupMenu.destroy();
-        this.$super();
+        if (this.__displayValue) this.__displayValue.destroy()
+        if (this.__popupMenu) this.__popupMenu.destroy()
+        this.$super()
```
[the remark this answers is not in the document]
Comment (Rincelent) [reply src/composite/select.js]: Bien vu c'esy corriger.

**Comment by flozz** [comment]
The changes looks good so far. There is only some coding style issue:

```
$ grunt jshint
Running "jshint:all" (jshint) task

   src/composite/select.js
    117 |        if (this.__displayValue) this.__displayValue.destroy();
                                          ^ Expected '{' and instead saw 'this'.
    329 |        if (this.__displayValue) this.__displayValue.destroy();
                                          ^ Expected '{' and instead saw 'this'.
    330 |        if (this.__popupMenu) this.__popupMenu.destroy();
                                       ^ Expected '{' and instead saw 'this'.

>> 3 errors in 54 files
```

**Comment by flozz** [comment]
Thank you :)

**Pull request merged**